### ISSUE TITLE
clojure.core.match

### ISSUE THREAD

**Issue by rlp10** [babashka/babashka#594]
Would it be possible to include match from clojure.core.match?

Apologies if it's already there and I just can't work out how to find it!

**Comment by borkdude** [comment]
It's currently not there. I'll keep the issue open to see who is interested.

**Comment by rlp10** [comment]
It doesn't look like anyone else is interested, so perhaps it doesn't make sense to include it.

Would you be able to point me to any instructions showing how I could add it myself for a personal build?

**Comment by borkdude** [comment]
@rlp10 You can add it under a feature flag like a lot of other libraries already are. Check one of the `feature-xyz` directories and the build script for `BABASHKA_FEATURE_...` environment variables. Then you can contribute your changes back to babashka with the feature flag turned off by default. If you need any help, let me know.

**Comment by borkdude** [comment]
Also see https://github.com/borkdude/babashka/blob/master/doc/build.md#feature-flags

**Comment by rlp10** [comment]
That's great, thank you. I'll see if I can work it out. I'm not very good at this stuff, but I really want match! :)

**Comment by borkdude** [comment]
There may be a challenge with getting the macros in, but I'm fairly sure it's possible. Luckily there are already plenty of examples and I'm willing to help you out, so we will get there eventually :).

**Comment by borkdude** [comment]
@rlp10 Another matching library does seem to work with babashka:

https://github.com/borkdude/babashka/issues/631

**Comment by 15joeybloom** [comment]
Just chiming in to say I'm also interested in using `clojure.core.match` with babashka!

**Comment by borkdude** [comment]
I've prepared the bindings in bb:

macOS: https://14436-201467090-gh.circle-artifacts.com/0/release/babashka-0.2.7-SNAPSHOT-macos-amd64.zip
linux: https://14440-201467090-gh.circle-artifacts.com/0/release/babashka-0.2.7-SNAPSHOT-linux-amd64.zip

Demo:

```
$ ./bb -e "(require '[clojure.core.match :as m]) (m/match [0 1 2 3] [0 a b 3] [a b])"
[1 2]
```

Some testing would be nice.

**Comment by didibus** [comment]
I think the decision here is: does Babashka differ to Clojure in that it wants to encourage the use of pattern matching as "standard" when Clojure does not?

See: https://www.reddit.com/r/Clojure/comments/gee6m5/just_found_out_why_clojure_doesnt_have_pattern/

**Comment by borkdude** [comment]
Another good point @didibus mentioned in the bb slack channel: this library isn't that widely used, according to a Github search.

Also nobody mentioned it as a missing feature in the last babashka survey:

https://github.com/borkdude/babashka/blob/master/doc/surveys/2020-11.md

Other libs like clojure.spec may have more priority and even this hasn't happened yet (see https://github.com/babashka/babashka/issues/558). 

So I'll proceed adding this as a feature flag, until I see more compelling evidence/examples that core.match is a good fit for bb.

**Comment by borkdude** [comment]
Another experiment might be to create a bb-compatible fork of core.match so you can run it from source. 

The `match` macro emits code that is interpreted by bb anyway (no direct native compilation possible), so that must be feasible somehow.

**Comment by borkdude** [comment]
![Screenshot 2021-01-18 at 16 47 10](https://user-images.githubusercontent.com/284934/104936493-e5d32880-59ac-11eb-8bb9-c03fac1cf9ae.png)

**Comment by borkdude** [comment]
Decided to go ahead and merge the branch, since I consider this feature upvoted enough.

**Comment by borkdude** [comment]
Now released with babashka 0.2.8.
